core/node/consensus/src/storage/store.rs

```diff
@@ -175,14 +178,23 @@ impl StoreRunner {
                 // Loop updating `blocks_persisted` whenever blocks get pruned.
                 const POLL_INTERVAL: time::Duration = time::Duration::seconds(1);
                 loop {
-                    let range = pool
-                        .connection(ctx)
-                        .await?
-                        .block_certificates_range(ctx)
-                        .await
-                        .wrap("block_certificates_range()")?;
-                    blocks_persisted.update(range);
-                    ctx.sleep(POLL_INTERVAL).await?;
+                    let span = tracing::info_span!("blocks_persisted_loop");
+                    async {
+                        let range = pool
+                            .connection(ctx)
+                            .await?
+                            .block_certificates_range(ctx)
+                            .await
+                            .wrap("block_certificates_range()")?;
+                        blocks_persisted.update(range);
+                        ctx.sleep(POLL_INTERVAL)
+                            .instrument(tracing::info_span!("sleep"))
```

Comment:
Is this going to emit an INFO level log every time it enters/exits sleep? Do we really need this level of detail?

Unlike the potential sleeps involved in saving certificates, this just polls the DB to keep an in-memory data structure in sync with it. It is interesting to see those others as they indicate this node is behind others in some respect, but seeing logs from this every second would be a bit tedious.

Reply:
Fair point, removed

Reply:
Also, just to be clear: it won't produce a log but rather a span within a trace so shouldn't pollute the logs at all. That being said I don't think there is much merit to having this tracked.